integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/ProcessUtils.scala

```diff
@@ -32,17 +32,13 @@ object ProcessUtils extends Logging {
     val proc = pb.start()
     val outputLines = new ArrayBuffer[String]
 
-    Utils.tryWithResource(new InputStreamReader(proc.getInputStream)) { procOutput =>
-      Utils.tryWithResource(new BufferedReader(procOutput)) { (bufferedOutput: BufferedReader) =>
-        var line: String = null
-        do {
-          line = bufferedOutput.readLine()
-          if (line != null) {
-            logInfo(line)
-            outputLines += line
-          }
-        } while (line != null)
+    Utils.tryWithResource(Source.fromInputStream(proc.getInputStream, "UTF-8")) { output =>
+      for (line <- output.getLines) {
+        logInfo(line)
+        outputLines += line
       }
+    }{
+      output => output.close()
```

Comment:
Is calling `close` necessary?

Reply:
Yes, it is because `output` is not a subclass of `Closeable`. See the extra version of `Utils.tryWithResource` that I added for details.

Reply:
It seems `Source` is `Closeable`, https://www.scala-lang.org/api/current/scala/io/Source.html. So you don't need to call `close` explicitly.

Reply:
Ah, that is interesting. I think it became `Closeable` in scala version 2.12. `Source` in scala 2.11 is not `Closeable`. From https://github.com/scala/scala/blob/v2.11.8/src/library/scala/io/Source.scala#L190:

```
abstract class Source extends Iterator[Char] {
```

Reply:
Ah, OK.

Reply:
I actually managed to get rid of the close call by using the input stream as resource. PTAL.

Reply:
LGTM.